hr_attendance_calendar/models/hr_holidays.py

```diff
@@ -1,39 +1,71 @@
 # -*- coding: utf-8 -*-
-# © 2016 Coninckx David (Open Net Sarl)
-# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
-
-from datetime import timedelta
-
-from openerp import models, fields, api
-import logging
-_logger = logging.getLogger(__name__)
-
-class HrHolidaysStatus(models.Model):
-    _inherit = 'hr.holidays.status'
-
-    remove_from_due_hours = fields.Boolean(string="Remove from due hours")
-    bonus_hours = fields.Boolean(string="Bonus hours")
-    
-    def name_get(self, cr, uid, ids, context=None):
-        res = super(HrHolidaysStatus, self).name_get(cr, uid, ids, context=context)
-        if not context.get('employee_id'):
-            return res
-        employee = self.pool.get('hr.employee').browse(cr, uid, context.get('employee_id'), context=context)[0]
-        res2 = []
-        for record in self.browse(cr, uid, ids, context=context):
-            if record.bonus_hours:
-                for name in res:
-                    if name[0] == record.id:
-                        name = (record.id, record.name + ('  (%0.2f h)' % (employee.bonus_malus)))
-                    res2.append(name)
-        return res2
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    @author: Stephane Eicher <seicher@compassion.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+
+import pytz
+
+from odoo import models, fields, api
+
 
 class HrHolidays(models.Model):
     _inherit = 'hr.holidays'
 
-    def _check_date(self, cr, uid, ids, context=None):
-        return True
+    ##########################################################################
+    #                                 FIELDS                                 #
+    ##########################################################################
+    hr_att_day_ids = fields.Many2many(comodel_name='hr.attendance.day',
+                                      string='Attendance day related')
+
+    ##########################################################################
+    #                             PUBLIC METHODS                             #
+    ##########################################################################
+    @api.multi
+    def write(self, vals):
+        super(HrHolidays, self).write(vals)
+        for att_call in self:
+            if att_call.state == 'validate' and 'hr_att_day_ids' not in vals:
+                start = fields.Date.from_string(att_call.date_from)
```

Comment:
You don't need to convert to date object because the search method takes strings as parameters (it's like a SQL query).